tasks/main.yml

```diff
@@ -9,42 +9,56 @@
     createhome: true
     home: "{{ caddy_home }}"
 
+- name: Build headers to use when making requests to github
+  set_fact:
+    caddy_github_headers: "{{ caddy_github_headers | combine({'Authorisation': 'token ' + caddy_github_token}) }}"
+  when: caddy_github_token | length > 0
+
 - name: Get all Caddy releases
   get_url:
     url: https://api.github.com/repos/mholt/caddy/git/refs/tags
     dest: "{{ caddy_home }}/releases.txt"
     force: true
+    headers: '{{ caddy_github_headers }}'
   retries: 3
   delay: 2
   when: caddy_update
   register: caddy_releases_cache
 
 - name: Set Caddy features
   copy:
-    content: "{{ caddy_features }}"
+    content: "{{ ','.join(caddy_packages) }}"
     dest: "{{ caddy_home }}/features.txt"
   when: caddy_update
   register: caddy_features_cache
 
+- name: Set some common parameters
+  set_fact:
+    caddy_bin_name: "caddy_{{ caddy_os }}_{{ go_arch }}"
+    caddy_common_args: "os={{ caddy_os }}&arch={{ go_arch }}&license={{ caddy_license }}\
```

Comment:
Maybe we can rename these? I don't understand why the args are `common`. Maybe `caddy_url_base` and `caddy_url_args`?

Reply:
Could we move these two variables into `defaults/main.yml` or does that not work for some reason?

Reply:
I have implemented both of these suggestions, thanks!

Reply:
This actually ended up a bit different now. The way it works now is that the whole `caddy_url` is set all at once in the vars file. If it is then decided to use gitlab the `caddy_url` is later changed using `set_fact` because the github url cannot be determined in the vars file - it relies on making a web request to github in order to learn the filename we need to request.